src/client/public/runner.ts

```diff
@@ -281,49 +291,70 @@ interface InitializationParams {
                     if (isListeningTo.snippetSwitching) {
                         $anotherSnippetSelected.find('.ms-MessageBar-text .snippet-name').text(input.message.name);
                         showReloadNotification($anotherSnippetSelected,
-                            () => clearAndRefresh(input.message.id, input.message.name),
+                            () => clearAndRefresh(input.message.id, input.message.name, false /*isTrustedSnippet*/),
                             () => isListeningTo.snippetSwitching = false);
                     }
                 }
             });
 
-        heartbeat.messenger.listen<ISnippet>()
+        heartbeat.messenger.listen<{ snippet: ISnippet, isTrustedSnippet: boolean }>()
             .filter(({ type }) => type === MessageType.REFRESH_RESPONSE)
             .subscribe(input => {
-                const snippet = input.message;
+                const snippet = input.message.snippet;
                 const data = JSON.stringify({
                     snippet: snippet,
                     returnUrl: returnUrl
                 });
 
                 // Use jQuery post rather than the Utilities post here
                 // (don't want to navigate, just to do an AJAX call)
-                $.post(window.location.origin + '/compile/snippet', { data: data })
-                    .then(html => processSnippetReload(html, snippet))
+                $.post(window.location.origin + '/compile/snippet', { data: data, isTrustedSnippet: input.message.isTrustedSnippet })
+                    .then(html => processSnippetReload(html, snippet, input.message.isTrustedSnippet))
                     .fail(handleError)
                     .always(() => {
-                        $('.runner-overlay').hide();
-                        $('.runner-notification').not('#notify-error').hide();
+                        if (!notifySnippetNotTrustedDisplayed) {
+                            $('.runner-overlay').hide();
+                            $('.runner-notification').not('#notify-error').hide();
+                        }
                     });
             });
     }
 
     function showReloadNotification($notificationContainer: JQuery, reloadAction: () => void, dismissAction: () => void) {
-        $notificationContainer.find('.action-fast-reload').off('click').click(reloadAction);
+        /* Do not display other messages while the trust message is displayed */
+        if (notifySnippetNotTrustedDisplayed && $notificationContainer.attr('id') !== 'notify-snippet-not-trusted') {
+            return;
+        }
+
+        $notificationContainer.find('.action-fast-reload').off('click').click(() => {
+            if ($notificationContainer.attr('id') === 'notify-snippet-not-trusted') {
```

Comment:
There is probably a better way then querying the DOM through the application state.